### ISSUE TITLE
Newrelic_rpm gem (9.2.1) displaying errors in rails console

### ISSUE THREAD

**Issue by JulienDefrance** [newrelic/newrelic-ruby-agent#1981]
## Description
Upgrading to the latest version of the newrelic_rpm gem (9.2.1) resulted in errors being printed in console.

```
#<Thread:0x00000001122d1820 /.../gems/newrelic_rpm-9.2.1/lib/new_relic/agent/tracer.rb:425 run> terminated with exception (report_on_exception is true):
/.../gems/newrelic_rpm-9.2.1/lib/new_relic/agent/transaction.rb:443:in `finished?': undefined method `finished?' for nil:NilClass (NoMethodError)

        initial_segment.finished?
                       ^^^^^^^^^^
	from /.../gems/newrelic_rpm-9.2.1/lib/new_relic/agent/tracer.rb:427:in `block in thread_block_with_current_transaction'
method=GET path=/documents/1 format=html controller=DocumentsController action=show status=200 duration=8456.51 view=71.72 db=0.00
#<Thread:0x00000001122bf300 /.../gems/newrelic_rpm-9.2.1/lib/new_relic/agent/tracer.rb:425 run> terminated with exception (report_on_exception is true):
#<Thread:0x00000001122d1820 /.../gems/newrelic_rpm-9.2.1/lib/new_relic/agent/tracer.rb:425 run> terminated with exception (report_on_exception is true):
/.../gems/newrelic_rpm-9.2.1/lib/new_relic/agent/transaction.rb:443:in `finished?': undefined method `finished?' for nil:NilClass (NoMethodError)

        initial_segment.finished?
                       ^^^^^^^^^^
	from /.../gems/newrelic_rpm-9.2.1/lib/new_relic/agent/tracer.rb:427:in `block in thread_block_with_current_transaction'
/.../gems/newrelic_rpm-9.2.1/lib/new_relic/agent/transaction.rb:443:in `finished?': undefined method `finished?' for nil:NilClass (NoMethodError)

        initial_segment.finished?
```

## Expected Behavior
No such error should be printed in console.

## Steps to Reproduce
- Update to newrelic_rpm 9.2.1
- Generate a request
- Look into the console output

## Your Environment
- Ruby 3.2.2
- Rails 7.0.4.3
- Puma 6.2.2
- newrelic_rpm gem (9.2.1)

## Additional context
- Recent changes added `finished?` method to the transaction class
- Adding safe navigation to https://github.com/newrelic/newrelic-ruby-agent/blob/dev/lib/new_relic/agent/transaction.rb (lines 438 through 445) might help resolve this issue.
- Define whether invoking .finished? on a transaction with no segment be returning true or false.

## For Maintainers Only or Hero Triaging this bug
*Suggested Priority (P1,P2,P3,P4,P5):*
*Suggested T-Shirt size (S, M, L, XL, Unknown):*

**Comment by JulienDefrance** [comment]
Cc: @tannalynn

**Comment by kaylareopelle** [comment]
Thanks for sharing this @JulienDefrance! I'm taking a look at this.

**Comment by kaylareopelle** [comment]
Hi @JulienDefrance, I have a fix available for this issue at #1983. We're preparing to release a new version of the agent that includes this change. 

If you'd like to test before the release, please update your Gemfile's reference for `newrelic_rpm` to:
```
gem 'newrelic_rpm', git: 'https://github.com/newrelic/newrelic-ruby-agent', branch: 'bugfix/finished-9-2-1'
```

**Comment by JulienDefrance** [comment]
Thank you!
